src/main/java/com/ericsson/eiffel/remrem/publish/helper/PublishUtils.java

```diff
@@ -0,0 +1,65 @@
+package com.ericsson.eiffel.remrem.publish.helper;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.eiffel.remrem.protocol.MsgService;
+import com.ericsson.eiffel.remrem.semantics.SemanticsService;
+import com.google.gson.JsonObject;
+
+import ch.qos.logback.classic.Logger;
+
+public class PublishUtils {
+
+    private static final String DOT = ".";
+    static Logger log = (Logger) LoggerFactory.getLogger(PublishUtils.class);
+    /**
+     * Method returns the MsgService based on the mp(message protocol) from the list of MsgService beans. 
+     * @param mp(message protocol) Specifies which service we consider from the list of MsgService beans
+     * @param msgServices List of msgService beans or instances
+     * @return msgService or null if message service not available.
+     */
+    public static MsgService getMessageService(String mp, MsgService msgServices[]) {
+        if (StringUtils.isNotBlank(mp)) {
+            for (MsgService service : msgServices) {
+                if (service.getServiceName().equals(mp)) {
+                    return service;
+                }
+            }
+        }
+        else {
+            for (MsgService service : msgServices) {

```

Comment:
If no mp parameter is specified use the first in the list. Then we are prepared for  dynamic loading of jars.